tests/worker/test_workflow.py

```diff
@@ -2097,6 +2097,30 @@ async def status() -> str:
         # TODO(cretz): Do more specific checks once we clean up traces
         assert "never_completing_coroutine" in trace
 
+async def test_workflow_enhanced_stack_trace(client: Client):
+    async with new_worker(
+        client, StackTraceWorkflow, LongSleepWorkflow, activities=[wait_cancel]
+    ) as worker:
+        handle = await client.start_workflow(
+            StackTraceWorkflow.run,
+            id=f"workflow-{uuid.uuid4()}",
+            task_queue=worker.task_queue,
+        )
+
+        # Wait until waiting
+        async def status() -> str:
+            return await handle.query(StackTraceWorkflow.status)
+
+        await assert_eq_eventually("waiting", status)
+
+        # Send stack trace query
+        trace = await handle.query("__enhanced_stack_trace")
+        
+        assert "never_completing_coroutine" in [ stack['functionName'] for stack in trace['stacks'] ]
+        # first line of never_completing_coroutine
+        assert 'self._status = "waiting"' in str(trace['sources'])
+        assert trace['sdk']['version'] == __version__
```

Comment:
Cool!

At least one other thing to test that I think is probably pretty important is to have the workflow run a couroutine that is defined in a different file, so that we can see the other file gets loaded properly

Reply:
Added an external file to helpers and created integration test to ensure it's included.